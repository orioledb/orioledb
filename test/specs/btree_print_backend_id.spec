setup
{
	CREATE EXTENSION IF NOT EXISTS orioledb;
	CREATE TABLE IF NOT EXISTS o_test
	(
		id integer NOT NULL,
		val text,
		PRIMARY KEY(id)
	) USING orioledb;
	TRUNCATE o_test;
}

teardown
{
	DROP TABLE o_test;
}

session "s1"
step "s1_begin" { BEGIN; }
step "s1_insert" { INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(1, 30, 5) id); }
step "s1_rollback" { ROLLBACK; }

session "s2"
step "s2_begin" { BEGIN; }
step "s2_insert" { INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(2, 30, 5) id); }
step "s2_rollback" { ROLLBACK; }

session "s3"
step "s3_begin" { BEGIN; }
step "s3_insert" { INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(3, 30, 5) id);
				   INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(5, 30, 5) id); }
step "s3_rollback" { ROLLBACK; }

session "s4"
step "s4_begin" { BEGIN; }
step "s4_insert" { INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(4, 30, 5) id); }
step "s4_rollback" { ROLLBACK; }

session "s_str"
step "s_str_structure" { SELECT orioledb_tbl_structure('o_test'::regclass, 'neb'); }

permutation "s1_begin" "s1_insert" "s2_begin" "s2_insert" "s3_begin" "s3_insert" "s4_begin" "s4_insert" "s_str_structure" "s1_rollback" "s2_rollback" "s3_rollback" "s4_rollback"
permutation "s1_insert" "s2_insert" "s3_begin" "s3_insert" "s4_begin" "s4_insert" "s_str_structure" "s3_rollback" "s4_rollback"
