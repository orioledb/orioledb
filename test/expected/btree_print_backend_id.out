Parsed test spec with 5 sessions

starting permutation: s1_begin s1_insert s2_begin s2_insert s3_begin s3_insert s4_begin s4_insert s_str_structure s1_rollback s2_rollback s3_rollback s4_rollback
step s1_begin: BEGIN;
step s1_insert: INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(1, 30, 5) id);
step s2_begin: BEGIN;
step s2_insert: INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(2, 30, 5) id);
step s3_begin: BEGIN;
step s3_insert: INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(3, 30, 5) id);
				   INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(5, 30, 5) id);
step s4_begin: BEGIN;
step s4_insert: INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(4, 30, 5) id);
step s_str_structure: SELECT orioledb_tbl_structure('o_test'::regclass, 'neb');
orioledb_tbl_structure                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_test_pkey contents
Page 0: level = 0, maxKeyLen = 8, nVacatedBytes = 0
state = free, datoid equal, relnode equal, ix_type = primary, dirty
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 256, hikey location = 72, hikey = ('13')
    Item 0: mode = update, backend = 0, offset = 280, tuple = ('1', '1val')
    Item 1: mode = update, backend = 1, offset = 320, tuple = ('2', '2val')
    Item 2: mode = update, backend = 2, offset = 360, tuple = ('3', '3val')
    Item 3: mode = update, backend = 3, offset = 400, tuple = ('4', '4val')
    Item 4: mode = update, backend = 2, offset = 440, tuple = ('5', '5val')
    Item 5: mode = update, backend = 0, offset = 480, tuple = ('6', '6val')
    Item 6: mode = update, backend = 1, offset = 520, tuple = ('7', '7val')
    Item 7: mode = update, backend = 2, offset = 560, tuple = ('8', '8val')
    Item 8: mode = update, backend = 3, offset = 600, tuple = ('9', '9val')
    Item 9: mode = update, backend = 2, offset = 640, tuple = ('10', '10val')
    Item 10: mode = update, backend = 0, offset = 680, tuple = ('11', '11val')
    Item 11: mode = update, backend = 1, offset = 720, tuple = ('12', '12val')
  Chunk 1: offset = 12, location = 760, hikey location = 80, hikey = ('21')
    Item 12: mode = update, backend = 2, offset = 776, tuple = ('13', '13val')
    Item 13: mode = update, backend = 3, offset = 816, tuple = ('14', '14val')
    Item 14: mode = update, backend = 2, offset = 856, tuple = ('15', '15val')
    Item 15: mode = update, backend = 0, offset = 896, tuple = ('16', '16val')
    Item 16: mode = update, backend = 1, offset = 936, tuple = ('17', '17val')
    Item 17: mode = update, backend = 2, offset = 976, tuple = ('18', '18val')
    Item 18: mode = update, backend = 3, offset = 1016, tuple = ('19', '19val')
    Item 19: mode = update, backend = 2, offset = 1056, tuple = ('20', '20val')
  Chunk 2: offset = 20, location = 1096, hikey location = 88
    Item 20: mode = update, backend = 0, offset = 1120, tuple = ('21', '21val')
    Item 21: mode = update, backend = 1, offset = 1160, tuple = ('22', '22val')
    Item 22: mode = update, backend = 2, offset = 1200, tuple = ('23', '23val')
    Item 23: mode = update, backend = 3, offset = 1240, tuple = ('24', '24val')
    Item 24: mode = update, backend = 2, offset = 1280, tuple = ('25', '25val')
    Item 25: mode = update, backend = 0, offset = 1320, tuple = ('26', '26val')
    Item 26: mode = update, backend = 1, offset = 1360, tuple = ('27', '27val')
    Item 27: mode = update, backend = 2, offset = 1400, tuple = ('28', '28val')
    Item 28: mode = update, backend = 3, offset = 1440, tuple = ('29', '29val')
    Item 29: mode = update, backend = 2, offset = 1480, tuple = ('30', '30val')

Index toast: not loaded

(1 row)

step s1_rollback: ROLLBACK;
step s2_rollback: ROLLBACK;
step s3_rollback: ROLLBACK;
step s4_rollback: ROLLBACK;

starting permutation: s1_insert s2_insert s3_begin s3_insert s4_begin s4_insert s_str_structure s3_rollback s4_rollback
step s1_insert: INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(1, 30, 5) id);
step s2_insert: INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(2, 30, 5) id);
step s3_begin: BEGIN;
step s3_insert: INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(3, 30, 5) id);
				   INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(5, 30, 5) id);
step s4_begin: BEGIN;
step s4_insert: INSERT INTO o_test (SELECT id, id || 'val' FROM generate_series(4, 30, 5) id);
step s_str_structure: SELECT orioledb_tbl_structure('o_test'::regclass, 'neb');
orioledb_tbl_structure                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_test_pkey contents
Page 0: level = 0, maxKeyLen = 8, nVacatedBytes = 0
state = free, datoid equal, relnode equal, ix_type = primary, dirty
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 256, hikey location = 72, hikey = ('13')
    Item 0: offset = 280, tuple = ('1', '1val')
    Item 1: offset = 320, tuple = ('2', '2val')
    Item 2: mode = update, backend = 0, offset = 360, tuple = ('3', '3val')
    Item 3: mode = update, backend = 1, offset = 400, tuple = ('4', '4val')
    Item 4: mode = update, backend = 0, offset = 440, tuple = ('5', '5val')
    Item 5: offset = 480, tuple = ('6', '6val')
    Item 6: offset = 520, tuple = ('7', '7val')
    Item 7: mode = update, backend = 0, offset = 560, tuple = ('8', '8val')
    Item 8: mode = update, backend = 1, offset = 600, tuple = ('9', '9val')
    Item 9: mode = update, backend = 0, offset = 640, tuple = ('10', '10val')
    Item 10: offset = 680, tuple = ('11', '11val')
    Item 11: offset = 720, tuple = ('12', '12val')
  Chunk 1: offset = 12, location = 760, hikey location = 80, hikey = ('21')
    Item 12: mode = update, backend = 0, offset = 776, tuple = ('13', '13val')
    Item 13: mode = update, backend = 1, offset = 816, tuple = ('14', '14val')
    Item 14: mode = update, backend = 0, offset = 856, tuple = ('15', '15val')
    Item 15: offset = 896, tuple = ('16', '16val')
    Item 16: offset = 936, tuple = ('17', '17val')
    Item 17: mode = update, backend = 0, offset = 976, tuple = ('18', '18val')
    Item 18: mode = update, backend = 1, offset = 1016, tuple = ('19', '19val')
    Item 19: mode = update, backend = 0, offset = 1056, tuple = ('20', '20val')
  Chunk 2: offset = 20, location = 1096, hikey location = 88
    Item 20: offset = 1120, tuple = ('21', '21val')
    Item 21: offset = 1160, tuple = ('22', '22val')
    Item 22: mode = update, backend = 0, offset = 1200, tuple = ('23', '23val')
    Item 23: mode = update, backend = 1, offset = 1240, tuple = ('24', '24val')
    Item 24: mode = update, backend = 0, offset = 1280, tuple = ('25', '25val')
    Item 25: offset = 1320, tuple = ('26', '26val')
    Item 26: offset = 1360, tuple = ('27', '27val')
    Item 27: mode = update, backend = 0, offset = 1400, tuple = ('28', '28val')
    Item 28: mode = update, backend = 1, offset = 1440, tuple = ('29', '29val')
    Item 29: mode = update, backend = 0, offset = 1480, tuple = ('30', '30val')

Index toast: not loaded

(1 row)

step s3_rollback: ROLLBACK;
step s4_rollback: ROLLBACK;
